src/redux/fallbackExplorer.js

```diff
@@ -221,51 +282,65 @@ const fetchAssetBalances = async (tokens, address, network) => {
   }
 };
 
-export const fallbackExplorerInit = () => async (dispatch, getState) => {
-  const { accountAddress, nativeCurrency, network } = getState().settings;
-  const { latestTxBlockNumber, mainnetAssets } = getState().fallbackExplorer;
-  const formattedNativeCurrency = toLower(nativeCurrency);
+export const fetchOnchainBalances = ({
+  keepPolling = true,
+  withPrices = true,
+}) => async (dispatch, getState) => {
+  logger.log('😬 FallbackExplorer:: fetchOnchainBalances');
+  const { network, accountAddress, nativeCurrency } = getState().settings;
+  const { assets: allAssets, genericAssets } = getState().data;
   const { coingeckoIds } = getState().additionalAssetsData;
-  // If mainnet, we need to get all the info
-  // 1 - Coingecko ids
-  // 2 - All tokens list
-  // 3 - Etherscan token transfer transactions
-  if (NetworkTypes.mainnet === network) {
-    const newMainnetAssets = await findAssetsToWatch(
-      accountAddress,
-      latestTxBlockNumber,
-      dispatch,
-      coingeckoIds
-    );
+  const formattedNativeCurrency = toLower(nativeCurrency);
+  const { mainnetAssets } = getState().fallbackExplorer;
+  const chainId = ethereumUtils.getChainIdFromNetwork(network);
+  const covalentMainnetAssets = await getMainnetAssetsFromCovalent(
+    chainId,
+    accountAddress,
+    AssetTypes.token,
+    formattedNativeCurrency,
+    coingeckoIds,
+    allAssets,
+    genericAssets
+  );
 
-    await dispatch({
+  const { assets: accountAssets } = getState().data;
+  let assets =
+    network === NetworkTypes.mainnet
+      ? uniqBy(
+          [...mainnetAssets, ...covalentMainnetAssets],
```

Comment:
`covalentMainnetAssets` can be null so we should do a check before doing destructuring